pkg/metricprovider/custom_metric_provider.go

```diff
@@ -240,11 +253,11 @@ func GetPrediction(ctx context.Context, kubeclient client.Client, namespace stri
 	err = kubeclient.List(ctx, predictionList, opts...)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get TimeSeriesPrediction when get custom metric ")
-	} else if len(predictionList.Items) != 1 {
-		return nil, fmt.Errorf("only one TimeSeriesPrediction should match the selector %s ", metricSelector.String())
+	} else if len(predictionList.Items) == 0 {
+		return nil, fmt.Errorf("there is no TimeSeriesPrediction match the selector %s ", metricSelector.String())
 	}
 
-	return &predictionList.Items[0], nil
+	return predictionList.Items, nil
```

Comment:
Why return a tsp list, is there some case that has more than one tsp for a metric query?

Reply:
When we query the TSP by effective-hpa-uid, we can achieve exact matching. However, when we query the TSP by Target, if the user misconfigures multiple ehpa, there may be multiple TSPS. In this case, matching by TSP and metricIdentifier is compatible. Subsequent processing can be handed over to the HPA